server/src/test/java/org/elasticsearch/action/explain/ExplainResponseTests.java

```diff
@@ -0,0 +1,131 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.explain;
+
+import org.apache.lucene.search.Explanation;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.get.GetResult;
+import org.elasticsearch.test.AbstractStreamableXContentTestCase;
+import org.elasticsearch.test.RandomObjects;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExplainResponseTests extends AbstractStreamableXContentTestCase<ExplainResponse> {
+    @Override
+    protected ExplainResponse doParseInstance(XContentParser parser) throws IOException {
+        return ExplainResponse.fromXContent(parser, randomBoolean());
+    }
+
+    @Override
+    protected ExplainResponse createBlankInstance() {
+        return new ExplainResponse();
+    }
+
+    @Override
+    protected ExplainResponse createTestInstance() {
+        String index = randomAlphaOfLength(5);
+        String type = randomAlphaOfLength(5);
+        String id = String.valueOf(randomIntBetween(1,100));
+        boolean exist = randomBoolean();
+        Explanation explanation = randomExplanation(randomExplanation(randomExplanation()), randomExplanation());
+        String fieldName = randomAlphaOfLength(10);
+        List<Object> values = Arrays.asList(randomAlphaOfLengthBetween(3, 10), randomInt(), randomLong(), randomDouble(), randomBoolean());
+        GetResult getResult = new GetResult(randomAlphaOfLengthBetween(3, 10),
+            randomAlphaOfLengthBetween(3, 10),
+            randomAlphaOfLengthBetween(3, 10),
+            randomNonNegativeLong(),
+            true,
+            RandomObjects.randomSource(random()),
+            singletonMap(fieldName, new DocumentField(fieldName, values)));
+        return new ExplainResponse(index, type, id, exist, explanation, getResult);
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return field -> field.startsWith("fields") || field.startsWith("get");
```

Comment:
I think that field.startsWith("get") could become field.startsWith("get._source") || field.startsWith("get.fields")

Reply:
Hi @javanna I change this line to `field.startsWith("get._source") || field.startsWith("get.fields")` but it still adds random items to "get.fields", it seems in this case it randomly inserts when field is "get" ?
Because when I do this `field.startsWith("get._source") || field.startsWith("get.fields") || field.equals("get")` then it works well.
So I keep this as `field.startsWith("get")` ?

Reply:
good point, that is because GetResult prints out fields also at the `get` level, depending on whether they are metadata fields or not. So any random field at the `get` level will be parsed back as a field and printed out as part of `fields`. 

You can use `field.equals("get") || field.startsWith("get.fields") || field.startsWith("get._source")`